Feat/game phases

naikibro opened the pull request:
## Changes
- created all required pages
- implemented the passing of the name from page to page
- the players now exist
- created a beautifull scoreboard
- created a player panel to show the player image

## TODO
- implement the `GameOverPage`
- handle the static global variables better as context ( turn, phase, scores... )
- merge correctly

REF: #20

naikibro committed:
#20 feat: add global assets

- created a Player class
- created a Game class to contain all game logic
- added player images

REF: #20

naikibro committed:
#20 feat: add player handling inside the Map and Scoreboard

- player is taken in consideration when displaying `scoreboard`
- player is taken in consideration when displaying the `MapComponent`

REF: #20

naikibro committed:
#20 feat: added `GameOverPage`

- all required pages exist
- passing the name from `LandingPage` to `WarfieldPage` now works flawlessly
- added action and event handling for changing between pages

TODO:
- complete the `GameOverPage`

REF: #20

naikibro commented:
[Screencast from 05-14-2024 10:55:58 AM.webm](https://github.com/naikibro/BatailleNavale/assets/64978347/efdc0159-1439-44d8-a90e-fea297337d5f)

naikibro committed:
#20 fix: merge, lost changes

- fixed lost changes during merge
- added 2 player handling in `MapComponent` constructor

TODO:
- complete the `GameOverPage`
- handle the clicks
- handle the turns
- handle the `Fleet` on actions
- make the WarfieldPage `MapComponent` refresh and revalidate on **next turn**

REF: #20

tteuru09 commented on gui/src/common/Game.java:
```diff
@@ -2,20 +2,18 @@
+        Game.isPlaying = false;
+        Game.phase = 0;
+        Game.turn = 0;
     }
```
C'est pas mieux de mettre "this.map" au lieu de "Game.map" ?

tteuru09 reviewed (commented):
Seems good to me ! except for the little comment i left.

naikibro committed:
#20 feat(access) make scoreboard static

- made scoreboard static to Game
- this way its easily accessible from anywhere
- added playerUpdate function to invert the current player playing
- added isser methods to Tile
- made MapComponent part of WarfieldPage.this so it can be modified in child components

## Next steps
- fix the Fleet
- make the buttons disappear after click

REF: #20

tteuru09 reviewed (commented):
Seems good to me
